netbox/dcim/graphql/types.py

```diff
@@ -67,447 +79,944 @@ class ComponentObjectType(
     """
     Base type for device/VM components
     """
-    class Meta:
-        abstract = True
+    _name: str
+    device: Annotated["DeviceType", strawberry.lazy('dcim.graphql.types')]
 
 
-class ComponentTemplateObjectType(
+@strawberry.type
+class ModularComponentType(ComponentType):
+    module: Annotated["ModuleType", strawberry.lazy('dcim.graphql.types')] | None
+
+
+@strawberry.type
+class ComponentTemplateType(
     ChangelogMixin,
     BaseObjectType
 ):
     """
     Base type for device/VM components
     """
-    class Meta:
-        abstract = True
+    _name: str
+    device_type: Annotated["DeviceTypeType", strawberry.lazy('dcim.graphql.types')]
+
 
+@strawberry.type
+class ModularComponentTemplateType(ComponentTemplateType):
+    """
+    Base type for ComponentTemplateModel which supports optional assignment to a ModuleType.
+    """
+    device_type: Annotated["DeviceTypeType", strawberry.lazy('dcim.graphql.types')] | None
+    module_type: Annotated["ModuleTypeType", strawberry.lazy('dcim.graphql.types')] | None
 
 #
 # Model types
 #
 
-class CableType(NetBoxObjectType):
-    a_terminations = graphene.List('dcim.graphql.gfk_mixins.CableTerminationTerminationType')
-    b_terminations = graphene.List('dcim.graphql.gfk_mixins.CableTerminationTerminationType')
-
-    class Meta:
-        model = models.Cable
-        fields = '__all__'
-        filterset_class = filtersets.CableFilterSet
 
-    def resolve_type(self, info):
-        return self.type or None
-
-    def resolve_length_unit(self, info):
-        return self.length_unit or None
+@strawberry_django.type(
+    models.CableTermination,
+    exclude=('termination_type', 'termination_id'),
```

Comment:
`exclude` needs to be a list to make type checking happy.
```suggestion
    exclude=['termination_type', 'termination_id'],
```